mm2src/coins/eth.rs

```diff
@@ -702,18 +703,19 @@ impl SwapOps for EthCoin {
         amount: BigDecimal,
         swap_contract_address: &Option<BytesJson>,
     ) -> TransactionFut {
-        let taker_addr = try_fus!(addr_from_raw_pubkey(taker_pub));
-        let swap_contract_address = try_fus!(swap_contract_address.try_to_address());
+        let taker_addr = try_tx_fus!(addr_from_raw_pubkey(taker_pub));
+        let swap_contract_address = try_tx_fus!(swap_contract_address.try_to_address());
 
         Box::new(
             self.send_hash_time_locked_payment(
```

Comment:
`EthCoin:: send_hash_time_locked_payment` should return `TransactionErr::TxRecoverableError` as well here :)